test/MinimalDelegation.execute.t.sol

```diff
@@ -517,4 +517,105 @@ contract MinimalDelegationExecuteTest is TokenHandler, DelegationHandler, Execut
+
+        calls = CallBuilder.init();
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 2e18)); // Transfer 2 tokenA
+
+        // Create hash of the calls + nonce and sign it
+        nonce++; // increment the nonce since first one hasnt been used yet
+        signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        hashToSign = signerAccount.hashTypedData(signedCalls.hash());
+        signature = signerTestKey.sign(hashToSign);
+
+        // Pack the execution data:
+        // 1. Encode the nonce and signature into opData
+        opData = abi.encode(nonce, signature);
+        // 2. Encode the calls and opData together
+        bytes memory executionData2 = abi.encode(calls, opData);
+
+        bytes[] memory executionDataArray = new bytes[](2);
+        executionDataArray[0] = executionData1;
+        executionDataArray[1] = executionData2;
+
+        // execute the batch of batches
+        signerAccount.execute(BATCH_OF_BATCHES_CALL, abi.encode(executionDataArray));
+        vm.snapshotGasLastCall("execute_BATCH_OF_BATCHES_twoBatches_oneCallEach");
+
+        // Verify the transfers succeeded
+        assertEq(tokenA.balanceOf(address(receiver)), 3e18);
+
+        // Verify the seq was incremented - sequence should increase by 2
+        assertEq(signerAccount.getSeq(0), 2);
+    }
+
+    function test_execute_batchOfBatches_revertsWithCallFailed() public {
+        Call[] memory calls = CallBuilder.init();
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 1e18)); // Transfer 1 tokenA
+
+        // Get the current nonce components for key 0
+        uint256 nonceKey = 0;
+        (uint256 nonce,) = _buildNextValidNonce(nonceKey);
+
+        // Create hash of the calls + nonce and sign it
+        SignedCalls memory signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        bytes32 hashToSign = signerAccount.hashTypedData(signedCalls.hash());
+        bytes memory signature = signerTestKey.sign(hashToSign);
+
+        // Pack the execution data:
+        // 1. Encode the nonce and signature into opData
+        bytes memory opData = abi.encode(nonce, signature);
+        // 2. Encode the calls and opData together
+        bytes memory executionData1 = abi.encode(calls, opData);
+
+        calls = CallBuilder.init();
+        // this call silently reverts because its over the balance
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 101e18));
+
+        // Create hash of the calls + nonce and sign it
+        nonce++; // increment the nonce since first one hasnt been used yet
+        signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        hashToSign = signerAccount.hashTypedData(signedCalls.hash());
+        signature = signerTestKey.sign(hashToSign);
+
+        // Pack the execution data:
+        // 1. Encode the nonce and signature into opData
+        opData = abi.encode(nonce, signature);
```

Comment:
& anotha wrap :)

this is tricky bc our calldata decoder doesnt revert rn... but this willl revert when u pull in main